charts/consul/templates/api-gateway-controller-clusterrole.yaml

```diff
@@ -245,11 +245,21 @@ rules:
   - patch
   - update
 {{- if .Values.global.enablePodSecurityPolicies }}
-- apiGroups: ["policy"]
-  resources: ["podsecuritypolicies"]
-  resourceNames:
-    - {{ template "consul.fullname" . }}-api-gateway-controller
+- apiGroups:
+  - policy
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
```

Comment:
This was previously narrowed to the specific `PodSecurityPolicy` created for the API Gateway controller; however, the controller needs to be able to create `Roles/RoleBindings` that use whatever `PSP` is named in the `GatewayClassConfig`. Since you can't grant permissions that you don't have, the controller needs to be able to use more than just the `PSP` that applies to itself.